lib/portlayer/exec/container.go

```diff
@@ -326,10 +337,8 @@ func (c *Container) start(ctx context.Context) error {
 	if c.vm == nil {
 		return fmt.Errorf("vm not set")
 	}
-	// get existing state and set to starting
-	// if there's a failure we'll revert to existing
-	finalState := c.updateState(StateStarting)
-	defer func() { c.updateState(finalState) }()
+	// Set state to Starting
+	c.updateState(StateStarting)
```

Comment:
We can use `SetState` here instead of `updateState` to get rid of the race here.